java/mykola-melnyk/src/com/zoolatech/lecture2/tasks/_2/SimpleRobot.java

```diff
@@ -0,0 +1,226 @@
+ * current location and direction.
+ * The north-west corner of a room is represented by coordinates (0,0);
+ * north-east - (W-1, 0); south-east - (W-1, H-1); south-west - (0, H-1).
+ * By default, a robot should be placed in the north-west corner facing south direction.
+ * However, it should be also possible to give any coordinates and direction as the
+ * initial robot location. Robot must not be able to get outside a room - in
+ * such case the robot should move as much as possible to the room border, print
+ * a warning message and stop. All next moves in the same direction should be prevented.
+ */
+
+import java.util.Scanner;
+
+enum Direction {   //set direction variants
+    WEST,
+    EAST,
+    NORTH,
+    SOUTH
+}
+
+public class SimpleRobot {
+    private int x = 0; // this will be a position on x-axis (along width)
+    private int y = 0;  // this will be a position along y-axis (along height)
+    private int W;  // this will be a width value
+    private int H;  // this will be a height value
+    private Direction dir;  // we set a direction
+    private String compass; //to convert enum value to human-readable output
+
+
+    { //Initialization block
+        Scanner scanner = new Scanner(System.in);
+
+        do {
+            System.out.println("Set the width of the room:");
+            W = scanner.nextInt();
+        } while (W <= 0);
+
+        do {
+            System.out.println("Set the height of the room:");
+            H = scanner.nextInt();
+        } while (H <= 0);
+
+        System.out.println("Default coordinates are 0:0. Set custom? (y/n)");
+        if (scanner.next().charAt(0) == 'y') {
+            do {
+                System.out.println("Enter x (width) axis coordinates " +
+                        "(must be > 0 and < room width):");
+                x = scanner.nextInt();
+            } while (x < 0 || x >= W);
+
+            do {
+                System.out.println("Enter y (height) axis coordinates " +
+                        "(must be > 0 and < room height):");
+                y = scanner.nextInt();
+            } while (y < 0 || y >= H);
+        }
+
+
+        System.out.println("Default direction is South. Set custom? (y/n)");
+        if (scanner.next().charAt(0) == 'y') {
+            System.out.println("Possible variants are: N, E, S, W");
+            if (scanner.next().charAt(0) == 'N') {
+                dir = Direction.NORTH;
+            } else if (scanner.next().charAt(0) == 'E') {
+                dir = Direction.EAST;
+            } else if (scanner.next().charAt(0) == 'S') {
+                dir = Direction.SOUTH;
+            } else if (scanner.next().charAt(0) == 'W') {
+                dir = Direction.WEST;
+            } else {
+                System.out.println("Unknown parameter. Setting to default (East).");
+            }
+        } else dir = Direction.EAST;  //set default dir to South
+
+        if (dir == Direction.WEST) {  //convert enum value to human-readable output
```

Comment:
same: since we use enums, we can optimize these blocks using the switch expression

Reply:
Resolved: rewrote the program code completely.